host-controller/src/main/java/org/jboss/as/host/controller/ServerInventoryImpl.java

```diff
@@ -802,18 +809,19 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
                         authorizeCallback.setAuthorized(authorizeCallback.getAuthenticationID().equals(authorizeCallback.getAuthorizationID()));
                     } else if (current instanceof PasswordCallback) {
                         ((PasswordCallback) current).setPassword(server.getAuthKey().toCharArray());
-                    } else if (current instanceof VerifyPasswordCallback) {
-                        VerifyPasswordCallback vpc = (VerifyPasswordCallback) current;
-                        vpc.setVerified(server.getAuthKey().equals(vpc.getPassword()));
-                    } else if (current instanceof DigestHashCallback) {
-                        DigestHashCallback dhc = (DigestHashCallback) current;
+                    } else if (current instanceof EvidenceVerifyCallback) {
+                        EvidenceVerifyCallback vpc = (EvidenceVerifyCallback) current;
+                        vpc.setVerified(server.getAuthKey().equals(vpc.applyToEvidence(PasswordGuessEvidence.class, e -> new String(e.getGuess()))));
+                    } else if (current instanceof CredentialCallback) {
+                        CredentialCallback dhc = (CredentialCallback) current;
                         try {
-                            UsernamePasswordHashUtil uph = new UsernamePasswordHashUtil();
-                            if (userName == null || realm == null) {
+                            if (realm == null) {
```

Comment:
It's fine. IDEA warns me that "username == null is always false" so ^^^ is just cruft removal.